src/platform/console/unix_console.cpp

```diff
@@ -27,33 +25,39 @@ namespace mp = multipass;
 
 namespace
 {
-std::atomic<bool> window_resized{false};
+ssh_channel global_channel;
 
-bool was_set(std::atomic<bool>& value)
+void change_ssh_pty_size(ssh_channel channel)
 {
-    bool old = value;
-    while (old && !value.compare_exchange_weak(old, false))
-    {
-    }
-    return old;
+    struct winsize win = {0, 0, 0, 0};
+
+    ioctl(fileno(stdout), TIOCGWINSZ, &win);
+    ssh_channel_change_pty_size(channel, win.ws_col, win.ws_row);
```

Comment:
This is not signal safe.
ssh_channel_change_pty_size can allocate.

Unfortunately, signal handlers are very restrained on what they can do without deadlocks or other concurrency issues.

The best you can do is avoid them all together and use a thread, see for example UnixSignalHandler in daemon_main.cpp which uses a combination of sigaddset, sigprocmask and sigwait to avoiding signal handlers entirely.